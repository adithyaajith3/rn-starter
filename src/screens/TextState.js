import React, { useState } from 'react';
import { StyleSheet, TextInput, View, Text } from 'react-native';


const TextState = () => {
    const [name, setName] = useState('');
    return <View style={styles.container}>
        <TextInput value={ name }
            style={ styles.text }
            autoCapitalize="none"
            autoCorrect={ false }
            onChangeText={ (inputValue) => setName(inputValue)} />
        <Text>My name is { name }</Text>
    </View>

}

styles = StyleSheet.create({ 
    text: {
        borderColor: 'black',
        borderWidth: 0.5
    },
    container: {
        flex: 1,
        backgroundColor: '#fff',
        padding: 20
    }

});

export default TextState;