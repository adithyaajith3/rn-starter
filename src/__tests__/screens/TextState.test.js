import React from 'react';
import renderer from 'react-test-renderer';

import TextState from '../../screens/TextState';

test('renders correctly', () => {
  const tree = renderer.create(<TextState />).toJSON();
  expect(tree).toMatchSnapshot();
});