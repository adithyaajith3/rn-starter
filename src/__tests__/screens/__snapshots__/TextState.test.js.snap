// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoCorrect={false}
    onChangeText={[Function]}
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "black",
        "borderWidth": 0.5,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <Text>
    My name is 
    
  </Text>
</View>
`;
